cpp/src/arrow/util/io_util.cc

```diff
@@ -97,10 +97,32 @@
 #include "arrow/util/logging.h"
 
 // For filename conversion
-#if defined(_WIN32)
+#ifdef _WIN32
 #include "arrow/util/utf8.h"
 #endif
 
+#ifdef _WIN32
+#include <psapi.h>
+#include <windows.h>
+
+#elif __unix__ || __unix || unix || (__APPLE__ && __MACH__)
+#include <sys/resource.h>
+
+#if __APPLE__ && __MACH__
+#include <mach/mach.h>
+
+#elif (_AIX || __TOS__AIX__) || (__sun__ || __sun || sun && (_SVR4 || __svr4__))
+#include <procfs.h>
```

Comment:
Does this branch actually get used in the implementation?

Reply:
it was unused and was removed.